tests/test_client.py

```diff
@@ -173,6 +177,25 @@ def create(self, stream, ext_obj=None, start_date=None):
             return [self.create_refunds(ext_obj, start_date).body.get('refund')]
         elif stream == 'payments':
             return [self.create_payments()]
+        elif stream == 'shifts':
+            employee_id = [employee['id'] for employee in self.get_all('employees')][0]
+            location_id = [location['id'] for location in self.get_all('locations')][0]
+
+            if not end_date:
+                start_date_parsed = singer.utils.strptime_with_tz(start_date)
+                end_date_datetime = start_date_parsed + timedelta(minutes = self.SHIFT_MINUTES)
+                end_date = singer.utils.strftime(end_date_datetime)
+
+            max_end_at = max([obj['end_at'] for obj in self.get_all('shifts')])
+            if start_date < max_end_at:
+                LOGGER.warning('Tried to create a Shift that overlapped another shift')
+                # Readjust start date and end date
+                start_date = max_end_at
+                start_date_parsed = singer.utils.strptime_with_tz(start_date)
+                end_date_datetime = start_date_parsed + timedelta(minutes = self.SHIFT_MINUTES)
+                end_date = singer.utils.strftime(end_date_datetime)
```

Comment:
We copied pasted this code. Can we refactor to DRY it up?

Reply:
Done